compiler/transf.nim

```diff
@@ -1019,7 +1019,9 @@ proc transform(c: PTransf, n: PNode): PNode =
     result = transformAddrDeref(c, n, {nkHiddenDeref})
   of nkAddr:
     result = transformAddrDeref(c, n, {nkDerefExpr, nkHiddenDeref})
-  of nkDerefExpr, nkHiddenDeref:
+  of nkDerefExpr:
+    result = transformAddrDeref(c, n, {nkAddr, nkHiddenAddr})
+  of nkHiddenDeref:
```

Comment:
I don't understand this change. At this point there should be no difference between `nkDerefExpr` and `nkHiddenDeref`.

Reply:
This transformation was done for `lent`'s sake. I think it's safe not to transform `nkDerefExpr`. After all, the old change didn't transform it as well. It was me that introduced the extra transformation for `nkDerefExpr`.

![image](https://github.com/nim-lang/Nim/assets/43030857/fa5eb313-5d34-4f89-ae79-bf5475ae78d7)

ref https://github.com/nim-lang/Nim/commit/b1fe1690c48fc3c24e3db2b2e2d12961f46f36f9

Reply:
At least, it should be much better than before, but I can also investigate it further.

Reply:
Oh, I see another solution.